File: Tree/countGoodNodesinBinaryTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    ans=0
    def solve(self,root: TreeNode, max_par:int):
        if root is not None:
            if root.val>=max_par:
                max_par=root.val
                self.ans=self.ans+1
            self.solve(root.left, max_par)
            self.solve(root.right, max_par)
    def goodNodes(self, root: TreeNode) -> int:
        self.solve(root, -10001)
        return self.ans

File: Tree/countGoodNodesinBinaryTree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int ans=0;
    public void solve(TreeNode root, int max_par){
        if(root==null) return ;
        if(root.val>=max_par){
            //System.out.println(root.val);
            max_par=root.val;
            ans++;
        }
        solve(root.left, max_par);
        solve(root.right, max_par);
    }

    public int goodNodes(TreeNode root) {
        solve(root, -10001);
        return ans;
    }
}